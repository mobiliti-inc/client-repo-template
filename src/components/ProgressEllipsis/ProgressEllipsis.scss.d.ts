// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'container': string;
  'dot': string;
  'dot--dark': string;
  'fades-in-out': string;
}
declare var cssExports: CssExports;
export = cssExports;
