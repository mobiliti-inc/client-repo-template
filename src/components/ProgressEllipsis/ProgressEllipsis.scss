@import "../../styles/_variables.scss";

.container {
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
}

.dot {
	height: 8px;
	width: 8px;
	margin-right: 12px;
	border-radius: 50%;
	background-color: $darker-blue;
	opacity: 0.1;
}

.dot--dark {
	composes: dot;
	opacity: 1;
}