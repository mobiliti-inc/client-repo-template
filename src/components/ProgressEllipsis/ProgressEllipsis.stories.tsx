import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import * as React from 'react';
import { ProgressEllipsis } from '../';

storiesOf('ProgressEllipsis', module)
	.add('default', () =>
		<div style={{ background: '#fff', width: "100px" }}><ProgressEllipsis step={text('step', 'password')} /></div>
	);
