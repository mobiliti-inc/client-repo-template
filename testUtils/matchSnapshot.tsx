const { shallowToJson } = require('enzyme-to-json');

export const matchSnapshot = (subject: any) => {
	expect(shallowToJson(subject)).toMatchSnapshot();
};
