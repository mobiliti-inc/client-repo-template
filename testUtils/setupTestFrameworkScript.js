const React = require("react");
// const CSSModules = require("react-css-modules");
const Enzyme = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

const { shallow, render, mount } = Enzyme;

Enzyme.configure({ adapter: new Adapter() });
jest.setTimeout(30000);

global.React = React;
// global.CSSModules = CSSModules;
global.shallow = shallow;
global.render = render;
global.mount = mount;
