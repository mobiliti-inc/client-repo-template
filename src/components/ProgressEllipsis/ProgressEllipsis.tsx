import * as React from "react";

import "./ProgressEllipsis.scss";

interface ProgressEllipsisProps {
	step: string;
}

const ProgressEllipsis: React.FC<ProgressEllipsisProps> = (props) => {
	const { step } = props;

	switch (step) {
		case "name":
			return (
				<div styleName="container">
					<div styleName="dot--dark" />
					<div styleName="dot" />
					<div styleName="dot" />
				</div>
			);
		case "email":
			return (
				<div styleName="container">
					<div styleName="dot" />
					<div styleName="dot--dark" />
					<div styleName="dot" />
				</div>
			);
		case "password":
			return (
				<div styleName="container">
					<div styleName="dot" />
					<div styleName="dot" />
					<div styleName="dot--dark" />
				</div>
			);
		default:
			return null;
	}
};

export default ProgressEllipsis;
