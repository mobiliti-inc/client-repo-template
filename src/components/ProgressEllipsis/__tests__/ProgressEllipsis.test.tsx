import React from 'react';

import { matchSnapshot } from '../../../../testUtils/matchSnapshot';
import { shallow } from 'enzyme';

import ProgressEllipsis from '../ProgressEllipsis';

describe('ProgressEllipsis', () => {
	it('renders name step correctly', () => {
		const subject = shallow(<ProgressEllipsis step="name" />);
		matchSnapshot(subject);
	});

	it('renders email step correctly', () => {
		const subject = shallow(<ProgressEllipsis step="email" />);
		matchSnapshot(subject);
	});

	it('renders password step correctly', () => {
		const subject = shallow(<ProgressEllipsis step="password" />);
		matchSnapshot(subject);
	});
});
