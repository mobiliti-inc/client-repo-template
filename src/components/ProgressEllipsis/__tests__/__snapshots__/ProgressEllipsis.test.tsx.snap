// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressEllipsis renders email step correctly 1`] = `
<div
  className="container"
>
  <div
    className="dot"
  />
  <div
    className="dot--dark dot"
  />
  <div
    className="dot"
  />
</div>
`;

exports[`ProgressEllipsis renders name step correctly 1`] = `
<div
  className="container"
>
  <div
    className="dot--dark dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
</div>
`;

exports[`ProgressEllipsis renders password step correctly 1`] = `
<div
  className="container"
>
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot--dark dot"
  />
</div>
`;
